### PR TITLE
Revert "gcc-arm-embedded: remove"

### LINKED ISSUE #51907: Build toolchain for bare-metal arm
## Issue description

I use Nix shells to setup the build environment for all my development projects. When targeting bare-metal arm platforms, it was possible until a few months ago to use `gcc-arm-embedded`. All would work as expected when passing `--specs=nano.specs` in the `LDFLAGS`.

Now that `gcc-arm-embedded` has been removed, the “way to go” seems to be using `pkgsCross.arm-embedded.buildPackages.gcc`. However, using this one the linker complains:

```
/nix/store/z6pvjan22wqxcg787kyw89jl64dhh2ks-arm-none-eabi-binutils-2.30/bin/arm-none-eabi-ld: cannot find -lc_nano
```

It seems `libc_nano` is part of `newlib-nano`, which is included in arm’s gcc distrubution. In the mean time, `pkgsCross.arm-embedded.buildPackages.gcc` depends on `newlib`, which is not the same.

### Steps to reproduce

1. Setup a project which targets  an arm MCU and uses newlib-nano
2. Setup a Nix shell with `pkgsCross.arm-embedded.buildPackages.gcc
3. Try to build the project
4. It fails, complaining about missing `libc_nano`

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.16.11-100.fc26.x86_64, Fedora, 26 (Workstation Edition)`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.1.2`
 - channels(***): `"home-manager, nixpkgs-19.03pre159614.f52b7f5c345"`
 - nixpkgs: `/home/***/.nix-defexpr/channels/nixpkgs`

### PR DESCRIPTION
**Pull request by matthewbauer**
This (fully) reverts commit c327df544ae0875ce71b0a82ae7a650a8e34a268.

Fixes #51907.

some people still need gcc-arm-embedded to be available.

/cc @ejpcmac

### CODE REVIEW

File: pkgs/development/compilers/gcc-arm-embedded/default.nix [hunk 1]
```diff
@@ -0,0 +1,50 @@
+{ stdenv, bzip2, patchelf, glibc, gcc, fetchurl, version, releaseType, sha256, ncurses
```
Comment (Ekleog) [inline pkgs/development/compilers/gcc-arm-embedded/default.nix]: This file is not used in this partial revert (it was used only pre-gcc6 before the removing commit).

I'd think either drop it, or re-add gcc<6 to all-packages?

**Comment by rasendubi** [comment]
I vote for a full revert, as some people may still use previous versions. (It is common for embedded projects to break after a gcc upgrade.)

**Comment by matthewbauer** [comment]
Yeah I'll just revert the whole thing.

**Comment by ejpcmac** [comment]
> I vote for a full revert, as some people may still use previous versions. (It is common for embedded projects to break after a gcc upgrade.)

Yes, this is indeed the case. We are currently using `gcc-arm-embedded-4_9` for a project. One of our dependencies has been tested with this version of GCC and we cannot take the risk to see some erratic behaviour due to some new optimisations or other differences in the binary generated by newer versions of GCC.

**Pull request merged**